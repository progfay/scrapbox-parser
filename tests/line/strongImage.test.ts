/* global describe it expect */

describe('strongImage', () => {
  it('Simple strong image', () => {
    expect(`[[http://example.com/image.png]]
[[https://example.com/image.JPG]]
[[https://example.com/image.svg]]
[[https://example.com/image.GIF]]`).toEqualWhenParsing([
      {
        indent: 0,
        type: 'line',
        nodes: [
          {
            type: 'strongImage',
            src: 'http://example.com/image.png'
          }
        ]
      },
      {
        indent: 0,
        type: 'line',
        nodes: [
          {
            type: 'strongImage',
            src: 'https://example.com/image.JPG'
          }
        ]
      },
      {
        indent: 0,
        type: 'line',
        nodes: [
          {
            type: 'strongImage',
            src: 'https://example.com/image.svg'
          }
        ]
      },
      {
        indent: 0,
        type: 'line',
        nodes: [
          {
            type: 'strongImage',
            src: 'https://example.com/image.GIF'
          }
        ]
      }
    ])
  })

  it('HTTP jpeg strong image with special and japanese chars', () => {
    expect('[[http://example.com/~!@#$%^&*()_+`-={}\\\'"?,.<>|/画像.jpeg]]').toEqualWhenParsing([
      {
        indent: 0,
        type: 'line',
        nodes: [
          {
            type: 'strongImage',
            src: 'http://example.com/~!@#$%^&*()_+`-={}\\\'"?,.<>|/画像.jpeg'
          }
        ]
      }
    ])
  })

  it('Gyazo image', () => {
    expect('[[https://gyazo.com/0f82099330f378fe4917a1b4a5fe8815]]').toEqualWhenParsing([
      {
        indent: 0,
        type: 'line',
        nodes: [
          {
            type: 'strongImage',
            src: 'https://gyazo.com/0f82099330f378fe4917a1b4a5fe8815/thumb/1000'
          }
        ]
      }
    ])
  })
})
