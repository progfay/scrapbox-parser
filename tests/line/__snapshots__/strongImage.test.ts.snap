// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`strongImage > Direct Gyazo image 1`] = `
[
  {
    "indent": 0,
    "nodes": [
      {
        "raw": "[[https://i.gyazo.com/0f82099330f378fe4917a1b4a5fe8815.png]]",
        "src": "https://i.gyazo.com/0f82099330f378fe4917a1b4a5fe8815.png",
        "type": "strongImage",
      },
    ],
    "type": "line",
  },
]
`;

exports[`strongImage > Gyazo image 1`] = `
[
  {
    "indent": 0,
    "nodes": [
      {
        "raw": "[[https://gyazo.com/0f82099330f378fe4917a1b4a5fe8815]]",
        "src": "https://gyazo.com/0f82099330f378fe4917a1b4a5fe8815/thumb/1000",
        "type": "strongImage",
      },
    ],
    "type": "line",
  },
]
`;

exports[`strongImage > HTTP jpeg strong image with special and japanese chars 1`] = `
[
  {
    "indent": 0,
    "nodes": [
      {
        "raw": "[[http://example.com/~!@#$%^&*()_+\`-={}\\'"?,.<>|/画像.jpeg]]",
        "src": "http://example.com/~!@#$%^&*()_+\`-={}\\'"?,.<>|/画像.jpeg",
        "type": "strongImage",
      },
    ],
    "type": "line",
  },
]
`;

exports[`strongImage > Simple strong image 1`] = `
[
  {
    "indent": 0,
    "nodes": [
      {
        "raw": "[[http://example.com/image.png]]",
        "src": "http://example.com/image.png",
        "type": "strongImage",
      },
    ],
    "type": "line",
  },
  {
    "indent": 0,
    "nodes": [
      {
        "raw": "[[https://example.com/image.JPG]]",
        "src": "https://example.com/image.JPG",
        "type": "strongImage",
      },
    ],
    "type": "line",
  },
  {
    "indent": 0,
    "nodes": [
      {
        "raw": "[[https://example.com/image.svg]]",
        "src": "https://example.com/image.svg",
        "type": "strongImage",
      },
    ],
    "type": "line",
  },
  {
    "indent": 0,
    "nodes": [
      {
        "raw": "[[https://example.com/image.GIF]]",
        "src": "https://example.com/image.GIF",
        "type": "strongImage",
      },
    ],
    "type": "line",
  },
  {
    "indent": 0,
    "nodes": [
      {
        "raw": "[[https://example.com/image.webp]]",
        "src": "https://example.com/image.webp",
        "type": "strongImage",
      },
    ],
    "type": "line",
  },
]
`;
